include/hip/hcc_detail/helpers.hpp

```diff
@@ -48,7 +48,24 @@ namespace hip_impl {
 template <typename...>
 using void_t_ = void;
 
-#if (__cplusplus < 201402L)
+#if ((__cplusplus >= 201703L) && defined(__cpp_lib_is_invocable))
+// C++17
+template <typename, typename = void>
+struct is_callable_impl : std::false_type {};
+
+template <FunctionalProcedure F, typename... Ts>
+struct is_callable_impl<F(Ts...), void_t_<std::invoke_result<F(Ts...)> > > : std::true_type {};
```

Comment:
`std::invoke_result<F(Ts...)>` is wrong. It should be `typename std::invoke_result<F, Ts...>::type` or `std::invoke_result_t<F, Ts...>` as shown [here](https://en.cppreference.com/w/cpp/types/result_of).

Reply:
Added alias to is_callable_impl with is_invocable

Reply:
A template alias wont work, because `is_callable` is used in hip as `is_callable<F(Ts...)>` and `is_invocable` is being called as `is_invocable<F, Ts...>`.